src/main/java/org/opentripplanner/transit/model/timetable/RealTimeTripTimes.java

```diff
@@ -263,7 +263,9 @@ public void setRealTimeState(final RealTimeState realTimeState) {
    * found.
    */
   public void validateNonIncreasingTimes() {
-    final int nStops = arrivalTimes.length;
+    final int nStops = arrivalTimes != null
+      ? arrivalTimes.length
+      : scheduledTripTimes.getNumStops();
```

Comment:
Why are you not using?
```suggestion
    final int nStops = scheduledTripTimes.getNumStops();
```

Reply:
This is indeed simpler since by construction:
* the array of  real-time arrival times is either null or has the same size as the array of scheduled arrival times
* the array of scheduled arrival times cannot be null (Exception thrown at construction time)  



Unrelated to this PR (should be fixed in a follow-up PR):
* Attempting to build  ScheduledStopTimes with a null array fails with NullPointerException instead of DataValidtionException
* It is possible to build  ScheduledStopTimes with arrival and departure time arrays of difference size, this should fail at construction time.